contracts/MedianOracle.sol

```diff
@@ -35,6 +35,7 @@ contract MedianOracle is Ownable, IOracle {
     event ProviderAdded(address provider);
     event ProviderRemoved(address provider);
     event ReportTimestampOutOfRange(address provider);
+    event ProviderReportPushed(address provider, uint256 payload, uint256 timestamp);
```

Comment:
@truncs Do any of these arguments need to be indexed?
What do you expect the query behavior to be like?

e.g.
would be need to query by provider?
query for payload outliers?
query for anything after a timestamp?

Reply:
> query for payload outliers?
> query for anything after a timestamp?
Indexing doesn't help with these two cases. 

Querying everything and processing after the fact is more practical that indexing and doing operations using the log query itself.

Reply:
Added indexing to provider.